hygieia-jenkins-plugin/src/main/java/hygieia/builder/SonarBuilder.java

```diff
@@ -73,21 +83,123 @@ private void setSonarDetails(AbstractBuild build, String buildId) throws IOExcep
+            // request to CE API is not required as Sonar Version < 5.2
+            return true;
+        }
 
+        // keep polling Sonar CE API for max configured attempts to fetch
+        // status of sonar analysis. After every attempt if CE API is not yet
+        // ready, sleep for configured interval period.
+        // Return true as soon as the status changes to SUCCESS
+        for(int i=0;i<this.ceQueryMaxAttempts;i++) {
+            // get the status of sonar analysis using Sonar CE API
+            RestCall.RestCallResponse ceAPIResponse = restCall.makeRestCallGet(this.sonarCEAPIUrl);
+            int responseCodeCEAPI = ceAPIResponse.getResponseCode();
+            if (responseCodeCEAPI == HttpStatus.SC_OK) {
+                String taskStatus = getCETaskStatus(ceAPIResponse.getResponseString());
+                switch (taskStatus) {
+                    case "IN_PROGRESS":
+                    case "PENDING":
+                        // Wait the configured interval then retry
+                        listener.getLogger().println("Waiting for report processing to complete...");
+                        try {
+                            Thread.sleep(this.ceQueryIntervalInSeconds * 1000);
+                        } catch (InterruptedException ie) {
+                            listener.getLogger().println("Sonar report processing errored while getting the status...");
+                            return false;
+                        }
+                        break;
+                    case "SUCCESS":
+                        // Exit
+                        listener.getLogger().println("Sonar report processing completed...");
+                        return true;
+                    default:
+                        listener.getLogger().println("Hygieia Publisher: Sonar CE API returned bad status: " + taskStatus);
+                        return false;
+                }
+
+            }
+            else {
+                listener.getLogger().println("Hygieia Publisher: Sonar CE API Connection failed. Response: " + responseCodeCEAPI);
+                return false;
+            }
+        }
+        listener.getLogger().println("Hygieia Publisher: Sonar CE API could not return response on time.");
+        return false;
     }
 
+
     public CodeQualityCreateRequest getSonarMetrics() throws ParseException {
         if (StringUtils.isEmpty(sonarServer) || StringUtils.isEmpty(sonarProjectID)) return null;
         String url = String.format(sonarServer + URL_METRIC_FRAGMENT, sonarProjectID, METRICS);
         RestCall restCall = new RestCall(publisher.getDescriptor().isUseProxy());
-        RestCall.RestCallResponse callResponse = restCall.makeRestCallGet(url);
-        int responseCode = callResponse.getResponseCode();
-        if (responseCode == HttpStatus.SC_OK) {
-            String resp = callResponse.getResponseString();
-            return buildQualityRequest(resp);
+
```

Comment:
In the old version, there is no concept of Compute Engine. Sonar processes the report during the analysis part itself and hence when the mvn sonar:sonar completes, old Sonar would already have generated the report. Thus, the Hygieia plugin can access theh Resources API and have the results.

In the new version, when the sonar:sonar goal is completed, Sonar exposes an endpoint for the report that has a Status element. Its value changes to Success once the report is available.